src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/AccessorBuilder.cs

```diff
@@ -0,0 +1,153 @@
+        private static MethodInfo s_buildSetAccessorInternal = typeof(FastInvokerBuilder).GetMethod(nameof(BuildSetAccessorInternal), BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+        private static MethodInfo s_make = typeof(FastInvokerBuilder).GetMethod(nameof(Make), BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static);
+
+        public static Func<object> GetMakeNewInstanceFunc(Type type)
+        {
+            Func<object> make = (Func<object>)s_make.MakeGenericMethod(type).CreateDelegate(typeof(Func<object>));
+            return make;
+        }
+
+        public static Getter BuildGetAccessor(MemberInfo memberInfo)
+        {
+            if (memberInfo is PropertyInfo)
+            {
+                var propInfo = (PropertyInfo)memberInfo;
+                Func<PropertyInfo, Getter> buildGetAccessorGeneric = (Func<PropertyInfo, Getter>)s_buildGetAccessorInternal.MakeGenericMethod(propInfo.DeclaringType, propInfo.PropertyType).CreateDelegate(typeof(Func<PropertyInfo, Getter>));
+                Getter accessor = buildGetAccessorGeneric(propInfo);
+                return accessor;
+            }
+            else if (memberInfo is FieldInfo)
+            {
+                return (obj) =>
+                {
+                    FieldInfo fieldInfo = (FieldInfo)memberInfo;
+                    var value = fieldInfo.GetValue(obj);
+                    return value;
+                };
+            }
+            else
+            {
+                throw new InvalidOperationException($"The type, {memberInfo.GetType()}, of memberInfo is not supported.");
+            }        
+        }
+
+        public static Setter BuildSetAccessor(MemberInfo memberInfo)
+        {
+            if (memberInfo is PropertyInfo)
+            {
+                PropertyInfo propInfo = (PropertyInfo)memberInfo;
+                if (propInfo.CanWrite)
+                {
+                    Func<PropertyInfo, Setter> buildSetAccessorGeneric = (Func<PropertyInfo, Setter>)s_buildSetAccessorInternal.MakeGenericMethod(propInfo.DeclaringType, propInfo.PropertyType).CreateDelegate(typeof(Func<PropertyInfo, Setter>));
+                    Setter accessor = buildSetAccessorGeneric(propInfo);
+                    return accessor;
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Property {propInfo.Name} of type {DataContract.GetClrTypeFullName(propInfo.DeclaringType)} cannot be set.");
+                }
+            }
+            else if (memberInfo is FieldInfo)
+            {
+                FieldInfo fieldInfo = (FieldInfo)memberInfo;
+                return (ref object obj, object val) =>
+                {
+                    fieldInfo.SetValue(obj, val);
```

Comment:
This should only ever get created once for any individual fieldInfo on a particular type. This isn't going to be done repeatedly.